### PR TITLE
Manually merge 2.12.x into 3.0.x

### PR DESCRIPTION
**Pull request by greg0ire**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -31,7 +31,8 @@
         {"name": "Jonathan Wage", "email": "jonwage@gmail.com"}
     ],
     "require": {
-        "php": "^7.3 || ^8.0",
+        "php": "^7.3 || ^8",
+        "ext-pdo": "*",
```
Comment (morozov) [inline composer.json:35]: This should be reverted. The change in the first line is non-essential, the addition of the second line is wrong.

File: src/Query/QueryBuilder.php [hunk 2]
```diff
@@ -3,6 +3,7 @@
 namespace Doctrine\DBAL\Query;
 
 use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Driver\ResultStatement;
```
Comment (morozov) [inline src/Query/QueryBuilder.php:6]: This looks like a mistake.

**Review by morozov — CHANGES_REQUESTED** [review]

File: src/Query/QueryBuilder.php (hunk 2 above)
Comment (greg0ire) [reply src/Query/QueryBuilder.php:6]: Indeed, and it's weird phpcs has nothing to complain about this :thinking:

File: composer.json (hunk 1 above)
Comment (SenseException) [reply composer.json]: Wrong for 3.0 or does this also affect 2.12?

Comment (morozov) [reply composer.json]: Only wrong for `3.0.x`.

**Review by morozov — APPROVED** [review]

**Pull request merged**